service/stacks/zephyr/sal_spp_interface.c

```diff
@@ -0,0 +1,842 @@
+    if (err < 0) {
+        BT_LOGE("Failed to create UUID: %d", err);
+        return err;
+    }
+
+    spp_client->sdp_discover.func = sdp_discover_cb;
+    spp_client->sdp_discover.pool = &spp_sdp_pool;
+    spp_client->sdp_discover.uuid = (const struct bt_uuid *)&spp_client->uuid_128;
+
+    err = bt_sdp_discover(spp_conn->conn, &spp_client->sdp_discover);
+    if (err < 0) {
+        BT_LOGE("Failed to discover service: %d", err);
+        return err;
+    }
+
+    return 0;
+}
+
+bt_status_t bt_sal_spp_connect(bt_address_t* addr, uint16_t conn_port, bt_uuid_t* uuid)
+{
+    sal_spp_connection_t* spp_conn;
+    struct bt_conn* conn;
+    uint16_t scn = PORT2SCN(conn_port);
+    char addr_str[BT_ADDR_STR_LENGTH] = { 0 };
+    char uuid_str[40] = { 0 };
+    sal_spp_client_t* spp_client;
+    int err;
+
+    if (!addr || scn > 30) {
+        BT_LOGE("Invalid parameters: addr=%p, scn=%d", addr, scn);
+        return BT_STATUS_PARM_INVALID;
+    }
+
+    bt_addr_ba2str(addr, addr_str);
+    bt_uuid_to_string(uuid, uuid_str, 40);
+    BT_LOGD("%s, addr:%s, conn_port:%d, uuid:%s", __func__, addr_str, conn_port, uuid_str);
+
+    spp_conn_lock();
+
+    /* check connection exists */
+    spp_conn = bt_list_find(g_spp_connections, spp_find_conn_by_port, &conn_port);
+    if (!spp_conn && (conn_port & 0x3F)) {
+        spp_conn = spp_find_device_by_dlci(addr, PORT2DLCI(conn_port, 0));
+    }
+
+    if (spp_conn) {
+        spp_conn_unlock();
+        BT_LOGE("SPP connection already exists for port %d", conn_port);
+        return BT_STATUS_BUSY;
+    }
+
+    spp_conn_unlock();
+
+    /* create spp connection object */
+    spp_conn = spp_connection_new((bt_address_t*)addr, conn_port, scn);
+    if (!spp_conn) {
+        BT_LOGE("Failed to allocate memory for SPP connection");
+        return BT_STATUS_NOMEM;
+    }
+
+    spp_client = zalloc(sizeof(sal_spp_client_t));
+    if (!spp_client) {
+        return BT_STATUS_NO_RESOURCES;
+    }
+
+    
+    /* find or create ACL connection */
+    conn = bt_conn_lookup_addr_br((bt_addr_t*)addr);
+    if (!conn) {
+        BT_LOGE("No ACL connection found for address: %s", addr_str);
+        spp_connection_free(spp_conn);
+        return BT_STATUS_FAIL;
+    }
+    
+    spp_conn->spp_client = spp_client;
+    spp_conn->conn = conn;
+    bt_conn_ref(spp_conn->conn);
+
+    if (conn_port & 0x3F) {
+        err = spp_connect_with_channel(spp_conn, scn);
+        if (err < 0) {
+            BT_LOGE("Failed to connect with scn: %d", err);
+            spp_connection_free(spp_conn);
+            return BT_STATUS_FAIL;
+        }
+    } else {
+        err = spp_connect_with_uuid(spp_conn, uuid);
+        if (err < 0) {
+            BT_LOGE("Failed to connect with uuid, err: %d", err);
+            spp_connection_free(spp_conn);
+            return BT_STATUS_FAIL;
+        }
+    }
+
+    spp_on_connection_state_changed((bt_address_t*)addr, conn_port, PROFILE_STATE_CONNECTING);
```

Comment:
move `spp_on_connection_state_changed((bt_address_t*)addr, conn_port, PROFILE_STATE_CONNECTING);` into `spp_connect_with_channel` and `sdp_discover_cb`.
```suggestion
    spp_on_connection_state_changed((bt_address_t*)addr, conn_port, PROFILE_STATE_CONNECTING);
```

Reply:
Do it in spp_connect_with_channel,and sdp would call it as well